static/app/views/replays/details.tsx

```diff
@@ -1,159 +1,66 @@
-  route: Props['route'];
-  router: Props['router'];
-};
-
-type State = AsyncView['state'];
-
-const EventHeader = ({event}: {event: Event}) => {
-  const message = getMessage(event);
-  return (
-    <EventHeaderContainer data-test-id="event-header">
-      <TitleWrapper>
-        <EventOrGroupTitle data={event} /> <FeatureBadge type="alpha" />
-      </TitleWrapper>
-      {message && (
-        <MessageWrapper>
-          <EventMessage message={message} />
-        </MessageWrapper>
-      )}
-    </EventHeaderContainer>
-  );
-};
-
-class ReplayDetails extends AsyncView<Props, State> {
-  state: State = {
-    loading: true,
-    reloading: false,
-    error: false,
-    errors: {},
-  };
-
-  getTitle() {
-    if (this.state.event) {
-      return `${this.state.event.id} - Replays - ${this.props.params.orgId}`;
-    }
-    return `Replays - ${this.props.params.orgId}`;
-  }
-
-  renderLoading() {
-    return <PageContent>{super.renderLoading()}</PageContent>;
-  }
-
-  renderBody() {
-    const {
-      location,
-      router,
-      route,
-      organization,
-      params: {eventSlug, orgId},
-    } = this.props;
-    return (
-      <ReplayLoader
-        eventSlug={eventSlug}
-        location={location}
-        orgId={orgId}
-        organization={organization}
-        router={router}
-        route={route}
-      />
-    );
-  }
-}
-
-function getProjectSlug(event: Event) {
-  return event.projectSlug || event['project.name']; // seems janky
-}
-
-// TODO(replay): investigate the `:fullscreen` CSS selector
-// https://caniuse.com/?search=%3Afullscreen
-const FullscreenWrapper = styled('div')<{isFullscreen: boolean}>`
-  ${p =>
-    p.isFullscreen
-      ? `
-    display: grid;
-    grid-template-rows: auto max-content;
-    background: ${p.theme.gray500};`
-      : ''}
-`;
+function ReplayDetails() {
+  const {
+    location,
+    params: {eventSlug, orgId},
+  } = useRouteContext();
 
-function ReplayLoader(props: ReplayLoaderProps) {
-  const orgSlug = props.orgId;
-  const {ref: fullscreenRef, isFullscreen, toggle: toggleFullscreen} = useFullscreen();
   const {
-    fetchError,
-    fetching,
-    onRetry,
     breadcrumbEntry,
     event,
-    replayEvents,
-    rrwebEvents,
     mergedReplayEvent,
-  } = useReplayEvent(props);
-
-  /* eslint-disable-next-line no-console */
-  console.log({
     fetchError,
     fetching,
     onRetry,
-    event,
-    replayEvents,
     rrwebEvents,
-    mergedReplayEvent,
+  } = useReplayEvent({
+    eventSlug,
+    location,
+    orgId,
   });
 
-  const renderMain = () => {
-    if (fetching) {
-      return <LoadingIndicator />;
-    }
-    if (!event) {
-      return <NotFound />;
-    }
+  const {ref: fullscreenRef, toggle: toggleFullscreen} = useFullscreen();
 
-    if (!rrwebEvents || rrwebEvents.length < 2) {
-      return (
+  if (fetching) {
+    return (
+      <DetailLayout event={event} orgId={orgId}>
+        <LoadingIndicator />
+      </DetailLayout>
```

Comment:
This is fine for now, but I think we'll want to think about using more skeleton placeholders when possible instead of a single loading indicator.